### PR TITLE
PatternAny by target source code and ust string

### LINKED ISSUE #160: Implement language-specific UST dump
If we convert the following PHP code

```php
$a->$b->$c
```

The following output from UST `.ToString()` we get:

```
a.b.c
```

It would be greate to use two modes: `Universal` and `Specific` for UST dump.

It can be used for more precise matching with `PatternAny` node.

Ideally it would be great to use [full-fidelity UST trees](https://github.com/dotnet/roslyn/wiki/Roslyn-Overview#syntax-trees).

### PR DESCRIPTION
**Pull request by KvanTTT**
* Correct RootUst text spans
* Match both by target source code and ust string, fixed #160 
* UstLinq: method names fixes (`Self` suffix)

### CODE REVIEW

File: Sources/PT.PM.Matching/PatternRoot.cs [hunk 1]
```diff
@@ -120,7 +120,7 @@ private static void TraverseChildren(PatternUst patternUst, Ust ust, MatchContex
                 }
 
                 var match = new MatchResult(
-                    ust is RootUst rootUst ? rootUst : ust.Root, context.PatternUst, context.Locations);
+                    ust.RootOrThis, context.PatternUst, context.Locations);
```
Comment (codeFather2) [inline Sources/PT.PM.Matching/PatternRoot.cs]: i think better way is use `ust.Root` property, and just override it in RootUst to `get => return this;`

File: Sources/PT.PM.CSharpParseTreeUst/Converter/CSharpRoslynParseTreeConverter.cs [hunk 2]
```diff
@@ -87,7 +87,10 @@ public override Ust Visit(SyntaxNode node)
 
             if (root == null)
             {
-                root = new RootUst(children.FirstOrDefault()?.Root?.SourceCodeFile, CSharp.Language);
```
Comment (codeFather2) [inline Sources/PT.PM.CSharpParseTreeUst/Converter/CSharpRoslynParseTreeConverter.cs]: why you decide to remove `SourceCodeFile` from here?

File: Sources/PT.PM.Matching/PatternRoot.cs (hunk 1 above)
Comment (KvanTTT) [reply Sources/PT.PM.Matching/PatternRoot.cs]: `RootUst` can have `Root` property itself in mixed-language files like JavaScript in PHP. In this case, JavaScript `RootUst` have a not null `Root` property that points to `RootUst` of PHP.

File: Sources/PT.PM.CSharpParseTreeUst/Converter/CSharpRoslynParseTreeConverter.cs (hunk 2 above)
Comment (KvanTTT) [reply Sources/PT.PM.CSharpParseTreeUst/Converter/CSharpRoslynParseTreeConverter.cs]: I've discovered that it's `null` in any case. Thus there is no sense in such expression.

Comment (codeFather2) [reply Sources/PT.PM.CSharpParseTreeUst/Converter/CSharpRoslynParseTreeConverter.cs]: so `RootUst` can include private field `root`, and property can look like `get => root ?? this`

File: Sources/PT.PM.Matching/PatternRoot.cs (hunk 1 above)
Comment (codeFather2) [reply Sources/PT.PM.Matching/PatternRoot.cs]: so RootUst can include private field root, and property can look like get => root ?? this.

**Pull request merged**